packages/compass-workspaces/src/stores/workspaces.ts

```diff
@@ -717,6 +812,11 @@ export const openWorkspace = (
       void dispatch(fetchCollectionInfo(workspaceOptions));
     }
 
+    if (workspaceOptions.type === 'Collections') {
+      // Fetching extra metadata for database should not block tab opening
+      void dispatch(fetchDatabaseInfo(workspaceOptions));
```

Comment:
Created [COMPASS-8897](https://jira.mongodb.org/browse/COMPASS-8897) to clean this up.